lib/superagent-mock.js

```diff
@@ -142,7 +181,20 @@ function mock (superagent, config, logger) {
       }
     }
 
-    if (typeof context.delay === 'number') {
+    if (context.progress && (isNodeServer ? this._formData : this.hasListeners && this.hasListeners('progress'))) {
```

Comment:
A ternary `if` in a `if`statement 😱 
Could you use a temporary variable instead ?